lib/nexpose/api_request.rb

```diff
@@ -97,27 +97,31 @@ def execute(options = {})
             end
           end
         end
-        # This is a hack to handle corner cases where a heavily loaded Nexpose instance
-        # drops our HTTP connection before processing. We try 5 times to establish a
-        # connection in these situations. The actual exception occurs in the Ruby
-        # http library, which is why we use such generic error classes.
-      rescue OpenSSL::SSL::SSLError => e
+      # This is a hack to handle corner cases where a heavily loaded Nexpose instance
+      # drops our HTTP connection before processing. We try 5 times to establish a
+      # connection in these situations. The actual exception occurs in the Ruby
+      # http library, which is why we use such generic error classes.
+      rescue OpenSSL::SSL::SSLError
         if @conn_tries < 5
           @conn_tries += 1
           retry
         end
-      rescue ::ArgumentError, ::NoMethodError => e
+      rescue ::ArgumentError, ::NoMethodError
         if @conn_tries < 5
           @conn_tries += 1
           retry
         end
       rescue ::Timeout::Error
+        $stdout.puts @http.read_timeout
         if @conn_tries < 5
           @conn_tries += 1
-          # If an explicit timeout is set, don't retry.
-          retry unless options.key? :timeout
+          if options.key?(:timeout)
+            retry if (time_tracker + options[:timeout].to_i) > Time.now
```

Comment:
This may not really be about this PR but more a general thought, should the retry here even occur now? Originally this was retried up to 5 times if no `:timeout` value was ever set, since there is now a default `:timeout` and the request passed it, would it make sense to simplify this now to just end at a single timeout either the default for user specified one?

Reply:
Scott and I were talking about not having retries at all anymore, maybe. At least as far as time out retries go, they are often more of a problem than just failing as many Nexpose API endpoints will do a lot of work on a heavy-load console before returning a response. Retrying in that case just repeats the work being done which we don't want to do IMO.